src/index.ts

```diff
@@ -33,17 +33,17 @@ const generateIv = (ivLengthInBits = ENCRYPTION_PARAMETERS.ivLength) => {
  * Generates the requisite randomised variables and initialises the cipher with them
  * @returns the cipher object, encryption key in hex, and iv in base64
  */
-const makeCipher = () => {
-  const encryptionKey = generateEncryptionKey();
+const makeCipher = (encryptionKey: string) => {
```

Comment:
i think can just put generateEncryptionKey() as a default param

Reply:
putting it as a default causing the problem when "" initialised at line 70 as it considered as a value hence no initialisation and if I remove that typescript throws error because it expects it to be string only not undefined.

Reply:
can do this default and set line70 one to be optional
http://dotnetpattern.com/typescript-optional-parameters

which would reflect the intent more clearly

Reply:
done